### DISCUSSION TITLE
Feature request: add a message (e.g., "If found, please phone...") to the cryptsetup lockscreen

### DISCUSSION THREAD

**Discussion by DeflateAwning** [linuxmint/discussions#365] [Ideas]
Is there a way to add a message to the login screen for full disk encryption (i.e., the cyptsetup one)? Which program handles that part of the boot process? 

If there's not a way to do it, it would be awesome if it could be configured easily through the regular UI.

**Comment by rcalixte** [comment]
This is software that is entirely upstream and not managed by the Mint team, similar to other software like the kernel or the web browsers. If you wanted something like this built in, your best bet would be to open an issue with the upstream project.

Until then, one alternative you could pursue would be using a custom image with the message or information you want displayed. You could even do something as simple as adding a banner to the existing image and using that. The software that controls this is called Plymouth. There are several pieces of documentation that you can use to accomplish this fairly easily without a GUI if you would like to spend the time on it:

* [Plymouth - ArchWiki](https://wiki.archlinux.org/title/plymouth)
* [plymouth - Debian Wiki](https://wiki.debian.org/plymouth)
* [Plymouth - Ubuntu Wiki](https://wiki.ubuntu.com/Plymouth)

**Comment by DeflateAwning** [comment]
That's helpful, thank you! Do you know where the "Please unlock disk <nmve...>" text is configured?

**Reply by rcalixte** [reply]
> That's helpful, thank you! Do you know where the "Please unlock disk <nmve...>" text is configured?

I'm not finding the string on a grep of my system so I presume it's from some compiled software, likely written in C. Check the sources for the related plymouth, cryptsetup, luks, etc. packages for the string. Modifying it means recompiling the software manually and then installing it but this isn't sustainable as any update will overwrite your changes.

Setting a custom image is likely the easiest route. You can modify any image (including the existing background image) and add a banner to your liking. The documentation links from before include steps for how to preview your custom background before applying it too. I imagine that it is a longer-term and more sustainable/maintainable solution.
